### DISCUSSION TITLE
error: 'stdarg.h' file not found [clang-diagnostic-error] when building on macOS

### DISCUSSION THREAD

**Discussion by GaryElshaw** [transmission/transmission#2835] [Q&A]
Fresh and clean git install, trying to compile with the CL w/ Xcode 13.3 installed.
[make-error.txt](https://github.com/transmission/transmission/files/8359218/make-error.txt)

This is a regularly occurring error during the make process as well.

[  4%] Completed 'natpmp'
[  4%] Built target natpmp
[  4%] Creating directories for 'miniupnpc'
[  4%] Performing download step (DIR copy) for 'miniupnpc'
Error copying directory from "/Users/gary/Documents/transmission/third-party/miniupnpc" to "/Users/gary/Documents/transmission/build/third-party/miniupnpc/src/miniupnpc".
make[2]: *** [third-party/miniupnpc/src/miniupnpc-stamp/miniupnpc-download] Error 1
make[1]: *** [CMakeFiles/miniupnpc.dir/all] Error 2
make: *** [all] Error 2

**Comment by Coeur** [comment]
Try this:
`git submodule foreach --recursive git clean -xfd`

It seems to have work for me regarding the `miniupnpc` issue.

**Reply by GaryElshaw** [reply]
> 4\. git submodule update --recursive

Sadly, i now get this, i've got this one before too.

--------------------------------------------------------

CMake Error at macosx/CMakeLists.txt:420 (add_executable):
  Cannot find source file:

    en.lproj/AddMagnetWindow.xib

  Tried extensions .c .C .c++ .cc .cpp .cxx .cu .mpp .m .M .mm .ixx .cppm .h
  .hh .h++ .hm .hpp .hxx .in .txx .f .F .for .f77 .f90 .f95 .f03 .hip .ispc


CMake Error at macosx/CMakeLists.txt:420 (add_executable):
  No SOURCES given to target: transmission-mac


CMake Generate step failed.  Build files cannot be regenerated correctly.

**Reply by Coeur** [reply]
Those are each time different issues.
If your source file is missing, then your git repository isn't properly cloned.
You may try (no guaranties):
```
git clean -xfd
git reset --hard
```

**Reply by GaryElshaw** [reply]
Back to this error again :-(

[ 15%] Building CXX object tests/gtest/CMakeFiles/gtestall.dir/__/__/third-party/googletest/googletest/src/gtest-all.cc.o
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/wchar.h:89:10: error: 'stdarg.h' file not found [clang-diagnostic-error]
#include <stdarg.h>
         ^~~~~~~~~~
1 error generated.
Error while processing /Users/gary/Transmission/third-party/googletest/googletest/src/gtest-all.cc.
Found compiler error(s).
make[2]: *** [tests/gtest/CMakeFiles/gtestall.dir/__/__/third-party/googletest/googletest/src/gtest-all.cc.o] Error 1
make[1]: *** [tests/gtest/CMakeFiles/gtestall.dir/all] Error 2
make: *** [all] Error 2

**Comment by Coeur** [comment]
It seems specific to googletest with your version of Xcode.
Can you try updating googletest?
```
cd third-party/googletest/
git checkout af29db7ec28d6df1c7f0f745186884091e602e07
```
If it works, then #2858 will resolve it.

**Reply by GaryElshaw** [reply]
I'm mostly using the CLI, i've given up on Xcode. I'll give that a go!

Unfortunately, no. It all looks very familiar doesn't it?

/googletest/googletest/src/gtest-all.cc.o
/Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/usr/include/wchar.h:89:10: error: 'stdarg.h' file not found [clang-diagnostic-error]
#include <stdarg.h>
         ^~~~~~~~~~
1 error generated.
Error while processing /Users/gary/Transmission/third-party/googletest/googletest/src/gtest-all.cc.
Found compiler error(s).
make[2]: *** [tests/gtest/CMakeFiles/gtestall.dir/__/__/third-party/googletest/googletest/src/gtest-all.cc.o] Error 1
make[1]: *** [tests/gtest/CMakeFiles/gtestall.dir/all] Error 2
make: *** [all] Error 2

**Comment by Coeur** [comment]
At this point, your issue doesn't seem to be in Transmission itself.
Try StackOverflow:
https://stackoverflow.com/questions/20206290/fatal-error-stdarg-h-file-not-found
https://stackoverflow.com/questions/46342411/wchar-h-file-not-found

First post suggests to redo your xcode-select.
Second post suggests "_to run CMake, 'Delete Cache', and 'Configure' the project again_" or do some symbolic linking.

**Reply by GaryElshaw** [reply]
Fresh install of the CLI SDK. I literally copied and pasted stdarg.h from one directory to another and got further along the make process. New error's now include, '/Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/stdbool.h:28:15: error: 'stdbool.h' file not found [clang-diagnostic-error]
#include_next <stdbool.h>'

stdbool is exactly where it should be. I'm curious to know why the 12.3 sdk is the default when the full list is...

MacOSX.sdk
MacOSX11.sdk
MacOSX12.sdk
MacOSX11.3.sdk
MacOSX12.0.sdk
MacOSX12.1.sdk
MacOSX12.3.sdk

Other fun things include:

list.cc.o
/Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/stdbool.h:28:15: error: 'stdbool.h' file not found [clang-diagnostic-error]
#include_next <stdbool.h>
              ^~~~~~~~~~~
/Users/gary/Transmission/libtransmission/announce-list.cc:33:24: warning: method 'remove' can be made static [readability-convert-member-functions-to-static]
bool tr_announce_list::remove(std::string_view announce_url)
                       ^
/Users/gary/Transmission/libtransmission/announce-list.cc:38:16: warning: redundant boolean literal in conditional return statement [readability-simplify-boolean-expr]
        return false;
~~~~~~~~~~~~~~~^~~~~~
/Users/gary/Transmission/libtransmission/announce-list.cc:45:24: warning: method 'remove' can be made static [readability-convert-member-functions-to-static]
bool tr_announce_list::remove(tr_tracker_id_t id)
                       ^
/Users/gary/Transmission/libtransmission/announce-list.cc:50:16: warning: redundant boolean literal in conditional return statement [readability-simplify-boolean-expr]
        return false;
~~~~~~~~~~~~~~~^~~~~~
/Users/gary/Transmission/libtransmission/announce-list.cc:57:24: warning: method 'replace' can be made static [readability-convert-member-functions-to-static]
bool tr_announce_list::replace(tr_tracker_id_t id, std::string_view announce_url_sv)
                       ^
/Users/gary/Transmission/libtransmission/announce-list.cc:75:24: warning: method 'add' can be made static [readability-convert-member-functions-to-static]
bool tr_announce_list::add(std::string_view announce_url_sv, tr_tracker_tier_t tier)
                       ^
/Users/gary/Transmission/libtransmission/announce-list.cc:202:37: warning: method 'getTier' can be made static [readability-convert-member-functions-to-static]
tr_tracker_tier_t tr_announce_list::getTier(tr_tracker_tier_t tier, tr_url_parsed_t const& announce) const
                                    ^                                                                ~~~~~
/Users/gary/Transmission/libtransmission/announce-list.cc:213:24: warning: method 'canAdd' can be made static [readability-convert-member-functions-to-static]
bool tr_announce_list::canAdd(tr_url_parsed_t const& announce)
                       ^
/Users/gary/Transmission/libtransmission/announce-list.cc:275:24: warning: method 'parse' can be made static [readability-convert-member-functions-to-static]
bool tr_announce_list::parse(std::string_view text)
                       ^
/Users/gary/Transmission/libtransmission/announce-list.cc:313:31: warning: method 'toString' can be made static [readability-convert-member-functions-to-static]
std::string tr_announce_list::toString() const
                              ^          ~~~~~
67699 warnings and 1 error generated.
Error while processing /Users/gary/Transmission/libtransmission/announce-list.cc.
Suppressed 67689 warnings (67689 in non-user code).
Use -header-filter=.* to display errors from all non-system headers. Use -system-headers to display errors from system headers as well.
Found compiler error(s).
make[2]: *** [libtransmission/CMakeFiles/transmission.dir/announce-list.cc.o] Error 1
make[1]: *** [libtransmission/CMakeFiles/transmission.dir/all] Error 2
make: *** [all] Error 2
gary@Garys-MacBook-Air build %
